### PR TITLE
Fix filling bar charts by data point

### PR DESCRIPTION
**Pull request by edkennard**
When using a HistogramRenderer with renderer.setDrawBars(true) and renderer.setPolyLineStyle(LineStyle.NONE), setting a fill colour on an individual data point per dataset.addDataStyle(index, "fillColor=cyan;") was not working.

The HistogramRenderer unit tests were already attempting to test this feature, but within the test environment the data point styles weren't being provided to the renderer correctly. This has also been addressed.

Reference_HistogramRendererTests_default

I've updated the reference screenshot accordingly, however I may need your guidance on whether that's appropriate because there are a couple of significant issues from me running on macOS:

- The fonts are rendered quite differently to the original reference snapshot
- I believe my localisation settings are causing the y-axis numbers to be formatted differently, UK style (0.50) instead of European (0,500).

I can see that FuzzyTestImageUtils.compareAndWriteReference does have some tolerance to differences, however I expect in this case I'm exceeding the threshold, in which case I might need your assistance to generate another reference snapshot which is closer and more meaningful.

### CODE REVIEW

**Review by wirew0rm — CHANGES_REQUESTED** [review]
Hey,
thanks for finding, fixing and testing this and also the detailed descriptions.

The Image looks fine, in fact the Locale we use for testing also doesn't use comma for the decimals. The configuration for the tests runner also shows how to change the locale: https://github.com/GSI-CS-CO/chart-fx/blob/master/pom.xml#L256
For the fonts the question is if it is the rendering or if we don't have the correct font available. Maybe we have to supply a compatible font which is available on mac? On the other hand it's imo not required to look pixel perfectly identical on each platform.
The test which test for image identity are only active if there is a reference image available and we didn't add those to the repository, because this would lead to difficult to debug CI features. So you can use this locally to check whether your changes produced visible differences, but for the CI it only serves as a smoke test which fails if there are NPEs or other uglinesses.
The idea would be to have the ci upload the offending images as artifacts and then we could review them manually and add them to the repository, but that is not implemented yet.

I got one small change request (see below), then we can merge this.

File: chartfx-dataset/src/main/java/de/gsi/dataset/testdata/spi/AbstractTestFunction.java
Comment (edkennard) [reply chartfx-dataset/src/main/java/de/gsi/dataset/testdata/spi/AbstractTestFunction.java]: Removing the function was the way forward

**Review by wirew0rm — APPROVED** [review]
Thank you for your patience, I'll merge it right away, didn't get to it before/during the weekend.

As for a release, it is well due, but there are a few issues that we wanted to address before. We are aiming for a release this week.
We also publish SNAPSHOT releases to maven central, but of course, if you want to release you have to release against a released version.

**Comment by edkennard** [comment]
@wirew0rm Thanks for getting this merged.

Regarding the release to Maven Central, for the moment the PR I've submitted to our project is using the workaround where I've included 3 manually built ChartFX JARs in a 'lib' folder, which SBT picks up automatically and everything works well. That PR is pending review for this week, and if we end up merging with that workaround then that's ok. Obviously using Maven Central is much preferred though, so I'll let the team know we expect a new ChartFX release sometime this week and I expect we'll hold my PR until that lands :)

**Comment by ennerf** [comment]
@edkennard I usually use [JitPack](https://jitpack.io/) when waiting for new releases. You can set the `<version>` to whatever commit hash you want, and it'll be picked up by Maven.

**Comment by edkennard** [comment]
@ennerf Thanks very much for this tip, I reckon I'll switch to this right now and get the JARs out of source control, then switch again to the official release as soon as that comes out.

I expect it'll be useful to know about this in the future with other GitHub projects, too

Cheers!

**Pull request merged**
